components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java

```diff
@@ -1125,7 +1125,13 @@ private void doUpdateGroups(String roleId, Set<String> newGroupIDList, Set<Strin
                         new ArrayList<>(deleteGroupIDList), tenantDomain);
             } catch (IdentityRoleManagementException e) {
                 if (RoleConstants.Error.INVALID_REQUEST.getCode().equals(e.getErrorCode())) {
-                    throw new BadRequestException(e.getMessage());
+                    // Error message and SCIM type
+                    String errorMessage = "Updating groups of the role by display name is not supported. "
+                            + "Update using group id instead.";
+                    String scimType = "invalidSyntax"; // From RFC 7644 Table 9
```

Comment:
Without hardcoding here you can directly use ResponseCodeConstants.INVALID_SYNTAX constant in charon

scimType variable is also redundant then